src/wp-includes/blocks/post-navigation-link.php

```diff
@@ -99,10 +99,24 @@ function render_block_core_post_navigation_link( $attributes, $content ) {
 		}
 	}
 
-	// The dynamic portion of the function name, `$navigation_type`,
-	// refers to the type of adjacency, 'next' or 'previous'.
+	$in_same_term = isset( $attributes['inSameTerm'] ) ? $attributes['inSameTerm'] : false;
+	$taxonomy     = isset( $attributes['taxonomy'] ) && $in_same_term ? $attributes['taxonomy'] : '';
+
+	/**
```

Comment:
```suggestion
	/*
```

Reply:
cc @carolinan 

Can you please follow-up with a PR in Gutenberg to address the comments that are related to the changes here